### PR TITLE
Update prompting logic if port is 0

### LINKED ISSUE #2075: Prompt if Port in Team Config is 0
**Is your feature or enhancement request related to a problem or limitation? Please describe**
With Zowe V3, some plug-ins include `port` in the template. Without a default value, it fills the port value with 0, which is invalid. This behavior stops prompting because a port is now specified. If the port is 0, which is both the `includeInTemplate` default and invalid, it should be ignored and prompted for

**Describe your enhancement idea**

Prompt for port if port is supplied as 0.
Prompt for host if host is supplied as "".

**Describe alternatives you've considered**

- Leave port as 0 and expect the end user to change it once the connection fails.

**Provide any additional context**

### PR DESCRIPTION
**Pull request by awharn**
**What It Does**

Prompts the user for port if the port retrieved from the user's config is 0 (default for includeInTemplate, invalid)
Resolves #2075 

**How to Test**

Install, set port to 0 in config for a service, observe prompting

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Review by anaxceron — APPROVED** [review]

**Review by adam-wolfe — APPROVED** [review]

File: packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts [hunk 1]
```diff
@@ -139,7 +139,8 @@ export class ConnectionPropsForSessCfg {
             promptForValues.push("hostname");
         }
 
-        if (ConnectionPropsForSessCfg.propHasValue(sessCfgToUse.port) === false && !doNotPromptForValues.includes("port")) {
+        if ((ConnectionPropsForSessCfg.propHasValue(sessCfgToUse.port) === false || sessCfgToUse.port === 0) &&
+            !doNotPromptForValues.includes("port")) {
```
Comment (zFernand0) [inline packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts:143]: clever way to prompt for port 🥳 

Question:
How would Zowe Explorer react to this new prompt item?

Curious if you could share a screenshot or something like that 😋 

Happy to approve afterwards this ⬆️

Comment (awharn) [reply packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts:143]: Zowe Explorer doesn't use AddPropsOrPrompt as far as Trae and I are aware. As for ZE, its current behavior is such that if a port is 0 in the team config, it tries using port 443 (for z/OSMF at least).

Comment (awharn) [reply packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts:143]: For the current CICS plug-in, a value of 0 results in an error.
image

Comment (zFernand0) [reply packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts:143]: thanks for providing more details around ZE (and related apps) 😋

**Review by zFernand0 — APPROVED** [review]
forgot to approve 😅

**Pull request merged**